### PR TITLE
feat(gatsby-sharp): create more resilient wrapper around sharp

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Creates a new package that wraps sharp and is 100% async. When sharp throws an error when we requiring it, we will rebuild sharp once to see if that fixes the issue. Sharp broken binary is one of the error most people hit. This happens when people switch between os architectures (Windows, WSL, M1, Rosetta, ...)

This is helpful as we will be importing sharp into Gatsby by default so we want to be as resilient as possible.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
     "npm-packlist": "^2.1.5",
     "npm-run-all": "4.1.5",
     "plop": "^1.9.1",
-    "prettier": "2.5.0",
+    "prettier": "^2.5.1",
```
Comment (wardpeet) [inline package.json:65]: It was necessary to fix a snapshot issue.

File: packages/gatsby-sharp/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    sharpInstance = sharp
+  }
+
+  return sharpInstance
+}
+
+function rebuildSharp(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    exec(
+      `npm rebuild sharp`,
```
Comment (LekoArts) [inline packages/gatsby-sharp/src/index.ts:35]: Does this also work when you use yarn 1/2?

Comment (wardpeet) [reply packages/gatsby-sharp/src/index.ts:35]: yes, I've been doing this trick for 2 years now :joy:.  As longs as npm gets shipped with nodejs we're good.

**Review by LekoArts — APPROVED** [review]

File: packages/gatsby-plugin-manifest/src/gatsby-node.js [hunk 3]
```diff
@@ -1,17 +1,13 @@
 import * as fs from "fs"
 import * as path from "path"
-import sharp from "./safe-sharp"
+// TODO V5 use gatsby/sharp
```
Comment (wardpeet) [inline packages/gatsby-plugin-manifest/src/gatsby-node.js:3]:
```suggestion
// TODO(v5): use gatsby/sharp
```

File: packages/gatsby-remark-images-contentful/src/index.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 const { selectAll } = require(`unist-util-select`)
-const sharp = require(`./safe-sharp`)
+// TODO V5 use gatsby/sharp
```
Comment (wardpeet) [inline packages/gatsby-remark-images-contentful/src/index.js:2]:
```suggestion
// TODO(v5): use gatsby/sharp
```

**Review by LekoArts — APPROVED** [review]

**Pull request merged**